src/catalog/default/default_functions.cpp

```diff
@@ -102,12 +102,12 @@ static const DefaultMacro internal_macros[] = {
 	{DEFAULT_SCHEMA, "array_to_string", {"arr", "sep", nullptr}, {{nullptr, nullptr}}, "list_aggr(arr::varchar[], 'string_agg', sep)"},
 	// Test default parameters
 	{DEFAULT_SCHEMA, "array_to_string_comma_default", {"arr", nullptr}, {{"sep", "','"}, {nullptr, nullptr}}, "list_aggr(arr::varchar[], 'string_agg', sep)"},
-	
+
 	{DEFAULT_SCHEMA, "generate_subscripts", {"arr", "dim", nullptr}, {{nullptr, nullptr}}, "unnest(generate_series(1, array_length(arr, dim)))"},
 	{DEFAULT_SCHEMA, "fdiv", {"x", "y", nullptr}, {{nullptr, nullptr}}, "floor(x/y)"},
 	{DEFAULT_SCHEMA, "fmod", {"x", "y", nullptr}, {{nullptr, nullptr}}, "(x-y*floor(x/y))"},
 	{DEFAULT_SCHEMA, "count_if", {"l", nullptr}, {{nullptr, nullptr}}, "sum(if(l, 1, 0))"},
-	{DEFAULT_SCHEMA, "split_part", {"string", "delimiter", "position", nullptr}, {{nullptr, nullptr}}, "coalesce(string_split(string, delimiter)[position],'')"},
+	{DEFAULT_SCHEMA, "split_part", {"string", "delimiter", "position", nullptr}, {{nullptr, nullptr}}, "if(string IS NOT NULL AND delimiter IS NOT NULL, coalesce(string_split(string, delimiter)[position],''), NULL)"},
```

Comment:
Postgres also returns `NULL`, if the position is `NULL`. Let's also align that: 
`if(string IS NOT NULL AND delimiter IS NOT NULL AND position IS NOT NULL, ...`


```sql
postgres=# select split_part('a,b,c','',null) is null;
 ?column? 
----------
 t
(1 row)
```

DuckDB
```sql
D select split_part('a,b,c','',null) is null;
┌─────────────────────────────────────────┐
│ (split_part('a,b,c', '', NULL) IS NULL) │
│                 boolean                 │
├─────────────────────────────────────────┤
│ false                                   │
└─────────────────────────────────────────┘
```